lib/pacemaker/pcmk_sched_primitive.c

```diff
@@ -486,13 +503,18 @@ pcmk__primitive_assign(pe_resource_t *rsc, const pe_node_t *prefer)
         }
         pe_rsc_info(rsc, "Unmanaged resource %s assigned to %s: %s", rsc->id,
                     (assign_to? assign_to->details->uname : "no node"), reason);
-        pcmk__assign_resource(rsc, assign_to, true);
+        pcmk__assign_resource(rsc, assign_to, true, stop_if_fail);
 
     } else if (pcmk_is_set(rsc->cluster->flags, pe_flag_stop_everything)) {
+        // Must stop at some point, but be consistent with stop_if_fail
+        if (stop_if_fail) {
+            pe_rsc_debug(rsc, "Forcing %s to stop: stop-all-resources",
+                         rsc->id);
+        }
         pe_rsc_debug(rsc, "Forcing %s to stop: stop-all-resources", rsc->id);
```

Comment:
This line was supposed to be dropped

I'm not tied to respecting `stop_if_fail` when `pe_flag_stop_everything` is set. It's less efficient this way. Seems less error-prone though, as code evolves.